naga/src/back/msl/writer.rs

```diff
@@ -885,6 +924,60 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
+    /// Writes the local variables of the given function, as well as any extra
+    /// out-of-bounds locals that are needed.
+    ///
+    /// The names of the OOB locals are also added to `self.names` at the same
+    /// time.
+    fn put_locals(&mut self, context: &ExpressionContext) -> BackendResult {
+        let oob_locals = context.oob_locals();
+        for &ty in oob_locals.iter() {
```

Comment:
style: Fun fact, you can shorten this to just a reference:

```suggestion
        for &ty in &oob_locals {
```

Reply:
Pushed as a `fixup!` with b5303f433f79ce888916d74837253b5bddea929a.

Reply:
There's actually a clippy lint for this, if we wanted to turn it on: https://rust-lang.github.io/rust-clippy/master/index.html#explicit_iter_loop